app/src/main/java/com/keylesspalace/tusky/components/conversation/ConversationsViewModel.kt

```diff
@@ -44,104 +46,117 @@ class ConversationsViewModel @Inject constructor(
         pagingSourceFactory = { database.conversationDao().conversationsForAccount(accountManager.activeAccount!!.id) }
     )
         .flow
+        .map { pagingData ->
+            pagingData.map { conversation -> conversation.toViewData() }
```

Comment:
Does this convert all of them each time any item changes? It's most likely okay, just for my understanding

Reply:
I think so, but i'm not 100% sure

Reply:
I'm afraid that it will be parsing HTML a lot in a few places

Reply:
I'll check how it actually works

Reply:
Ok I put a counter and a log statement into the mapping function, and it looks like there is some kind of working set that needs to be remapped on every revalidation. It is not the whole timeline but around 100 statuses. This is really not ideal as it also seems to run on the main thread. I need to dig deeper into this, thanks for pointing it out.

Reply:
Thanks for checking this! I think it's good that it's not the whole set and 100 statuses seems fine but main thread is not great ofc. Can we use `withContext()` or `flowOn()`?

Reply:
For some reason `flowOn` alone does not work, there needs to be an executor passed into the mapping and filter functions, which is weird. 

I will refactor it to filter before mapping in another PR.